src/config.js

```diff
@@ -0,0 +1,1603 @@
+          scopeDescriptor.getScopeChain(),
+          keyPath,
+          options
+        )
+      legacyScopeDescriptor = this.getLegacyScopeDescriptor(scopeDescriptor)
+      if (legacyScopeDescriptor) {
+        result.push(...Array.from(this.scopedSettingsStore.getAll(
+            legacyScopeDescriptor.getScopeChain(),
+            keyPath,
+            options
+          ) || []))
+      }
+    } else {
+      result = []
+    }
+
+    globalValue = this.getRawValue(keyPath, options)
+    if (globalValue) {
+      result.push({scopeSelector: '*', value: globalValue})
+    }
+
+    return result
+  }
+
+    // Essential: Sets the value for a configuration setting.
+    //
+    // This value is stored in Atom's internal configuration file.
+    //
+    // ### Examples
+    //
+    // You might want to change the themes programmatically:
+    //
+    // ```coffee
+    // atom.config.set('core.themes', ['atom-light-ui', 'atom-light-syntax'])
+    // ```
+    //
+    // You can also set scoped settings. For example, you might want change the
+    // `editor.tabLength` only for ruby files.
+    //
+    // ```coffee
+    // atom.config.get('editor.tabLength') # => 4
+    // atom.config.get('editor.tabLength', scope: ['source.ruby']) # => 4
+    // atom.config.get('editor.tabLength', scope: ['source.js']) # => 4
+    //
+    // # Set ruby to 2
+    // atom.config.set('editor.tabLength', 2, scopeSelector: '.source.ruby') # => true
+    //
+    // # Notice it's only set to 2 in the case of ruby
+    // atom.config.get('editor.tabLength') # => 4
+    // atom.config.get('editor.tabLength', scope: ['source.ruby']) # => 2
+    // atom.config.get('editor.tabLength', scope: ['source.js']) # => 4
+    // ```
+    //
+    // * `keyPath` The {String} name of the key.
+    // * `value` The value of the setting. Passing `undefined` will revert the
+    //   setting to the default value.
+    // * `options` (optional) {Object}
+    //   * `scopeSelector` (optional) {String}. eg. '.source.ruby'
+    //     See [the scopes docs](http://flight-manual.atom.io/behind-atom/sections/scoped-settings-scopes-and-scope-descriptors/)
+    //     for more information.
+    //   * `source` (optional) {String} The name of a file with which the setting
+    //     is associated. Defaults to the user's config file.
+    //
+    // Returns a {Boolean}
+    // * `true` if the value was set.
+    // * `false` if the value was not able to be coerced to the type specified in the setting's schema.
+  set () {
+    let [keyPath, value, options] = Array.from(arguments)
+
+    if (!this.settingsLoaded) {
+      this.pendingOperations.push(() => this.set(keyPath, value, options))
+    }
+
+    const scopeSelector = options != null ? options.scopeSelector : undefined
+    let source = options != null ? options.source : undefined
+    const shouldSave = (options != null ? options.save : undefined) != null ? (options != null ? options.save : undefined) : true
```

Comment:
It looks like `options != null ? options.save : undefined` is repeated twice here?

Reply:
Fixed!